podcast_downloader.py

```diff
@@ -66,19 +69,22 @@ def get_subtitles_by_track(
     tracks: List[Dict[str, Any]],
     lang: str = 'en',
     fmt: str = 'vtt',
+    directory=os.getcwd(),
```

Comment:
This works for this specific case but it's a little bit untraditional to see a function call setting a default parameter value in the function definition. It's probably better to use a pattern like this:

```python
def function(directory=None):
    directory = directory or os.getcwd()
```
The reason being, how you have it, the default value will get set once as soon as the file is imported (or for this case, when the script is evaluated by the Python interpreter before it starts executing). But, typically, there isn't a guarantee the function setting the default value will always return the same value, which could potentialy cause surprising behavior. Moving it inside the function means it will be called every time the parent function is called, so that the value returned will be correct depending on the current context.

Sticking with this specific example, python provides `os.chdir` which allows the script to change its working directory. Thus, if a future dev introduced a feature to change the current working directory into the desired output directory (which I should probably do haha), that would actually make the default value wrong, because it would've gotten set to the directory where the script was originally called from as opposed to where the current working directory was changed to.